app/components/Modals/SendModal/ConfirmDisplay.jsx

```diff
@@ -17,24 +19,60 @@ type Props = {
   net: NetworkType,
 }
 
-const ConfirmDisplay = ({
-  sendAddress,
-  sendAmount,
-  symbol,
-  confirmTransaction,
-  cancelTransaction,
-  explorer,
-  net
-}: Props) => (
-  <div>
-    <p>Please confirm the following transaction:</p>
-    <p>You are sending <strong>{formatBalance(symbol, sendAmount)} {symbol}</strong> to:</p>
-    <div className={styles.externalLink} onClick={() => openExplorerAddress(net, explorer, sendAddress)}>{sendAddress}</div>
-    <div>
-      <Button onClick={confirmTransaction}>Confirm</Button>
-      <Button cancel onClick={cancelTransaction}>Cancel</Button>
-    </div>
-  </div>
-)
+type State = {
+  addressChecked: boolean,
+  addressCheckedMessage: string
+}
+
+class ConfirmDisplay extends React.Component<Props, State> {
+  state = {
+    addressChecked: false,
+    addressCheckedMessage: ''
+  }
+
+  checkTransactionHistory = (net, address) => {
+    api.neonDB.getTransactionHistory(net, address).then((transactions) => {
```

Comment:
I believe you can do something like: (taken from transactions.js)

```
  let message = ''

  const [err, transactions] = await asyncWrap(api.neonDB.getTransactionHistory(net, address))
  if (err) {
  	message = 'Warning: recipient address has no activity in its transaction history. Please be sure the address is correct before sending.'
  	}
  } else if (!transactions || !transactions.length) {
      message = 'Warning: recipient address has no activity in its transaction history. Please be sure the address is correct before sending.'  	
  }

  this.setState({
  	addressChecked: true,
  	addressCheckedMessage: message
  })
```

This way you setState only once + use the async/await.
Also - `(net: NetworkType, address: string)`